grails-app/controllers/au/org/emii/portal/LayerController.groovy

```diff
@@ -117,4 +116,14 @@ class LayerController {
     def _isXmlContent(contentType) {
         return contentType.find(/(text|application)\/xml/)
     }
+
+    def _extractCharsetType(contentType) {
+        def returnedCharset
+        contentType.split(';').each { token ->
+            if (token.contains('charset')) {
```

Comment:
Go with:
```
if (token.contains('charset=')) {
```

Be as strict as possible here, especially when we're doing "non-standard" parsing.